base/array.jl

```diff
@@ -1721,10 +1722,10 @@ function findmax(a)
     mi = i = 1
     m, s = next(a, s)
     while !done(a, s)
+        m != m && break
         ai, s = next(a, s)
         i += 1
-        ai != ai && continue # assume x != x => x is a NaN
-        if m != m || isless(m, ai)
+        if ai != ai || isless(m, ai)
```

Comment:
if `ai != ai` I expect `isless` is not called. That shall cover the `NaN`- case. This can occur at most once, since the next loop circle is broken by the `m != m` condition few lines before.